underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -552,17 +535,63 @@ private boolean isDirectory(String path) throws IOException {
   }
 
   /**
-   * Lists the files or folders which match the given prefix.
+   * Lists the files or folders in the path and remove the path itself.
+   *
+   * @param path the prefix to match
+   * @param recursive whether to do a recursive listing
+   * @return a collection of the files or folders matching the prefix, or null if not found
+   * @throws IOException if path is not accessible, e.g. network issues
+   */
+  private String[] listHelper(String path, boolean recursive) throws IOException {
+    String prefix = addFolderSuffixIfNotPresent(stripContainerPrefixIfPresent(path));
```

Comment:
Given the meaning of `FolderSuffix` across under storages, this is a bit confusing. This line is really trying to accomplish `addPathSeparatorIfNotPresent` right?